### ISSUE TITLE
GPL - EPL licencing incompatibility for Clojure port

### ISSUE THREAD

**Issue by whilo** [qobi/R6RS-AD#1]
I am sorry to bother you with this. I am investigating the usage of autodifferentiation to provide efficent gradient descent updates for neural network training in Clojure. Unfortunately all the Scheme code is under GPL: sicmutils, r6rs-ad. Both libraries have ports for Clojure already (1)(2), but the licencing issue is worrying (3). Could you relicence R6RS-AD under e.g. LGPL or MIT? You could also provide an exception for EPL-based libraries to the GPL, to my knowledge (similar to the classpath exception of Java).

(1) https://github.com/littleredcomputer/sicmutils
(2) https://github.com/log0ymxm/clj-auto-diff 
(3) https://github.com/littleredcomputer/sicmutils/issues/31

**Comment by qobi** [comment]
Note that I am not the author/owner of SCMUTILS. You would need to contact
Gerry Sussman (gjs@mit.edu) for that.

I am the author/owner of R6RS-AD. I would be happy to relicense R6RS-AD under
both LPGL and MIT. I am also happy to provide an exception EPL-based libraries
to the GPL.

You can consider this email as executing such relicensing.

**Comment by whilo** [comment]
Thanks a lot! :)
